src/components/InputPanel.tsx

```diff
@@ -1,102 +1,171 @@
-import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
-import InputContext, { INPUT_TYPES } from '../contexts/InputContext';
-import styles from '../css/RMLMappingEditor.module.scss';
-import CodeEditor from './CodeEditor';
-import { ReactComponent as PlusIcon } from '../images/plus.svg';
-import { ReactComponent as DownArrow } from '../images/down-arrow.svg';
+import { useCallback, useContext, useEffect, useMemo, useState } from "react";
+import InputContext, { INPUT_TYPES } from "../contexts/InputContext";
+import styles from "../css/RMLMappingEditor.module.scss";
+import CodeEditor from "./CodeEditor";
+import { ReactComponent as PlusIcon } from "../images/plus.svg";
+import { ReactComponent as DownArrow } from "../images/down-arrow.svg";
 
 const views = {
-  inputs: 'Input Files',
-  // functions: 'Functions',
-}
+  inputs: "Input Files",
+  functions: "Functions",
+};
+type ViewType = keyof typeof views;
+
+const functionList = [
```

Comment:
Nice! I like the direction. I think each listed function should have a name, description, and list of arguments. Probably here in the left panel you would just see the name, then hovering over them or clicking a "more info" button would display an info box with the description and list of arguments.

We don't need the actual function implementations here because they are implemented in [@comake/rmlmapper-js](https://github.com/comake/rmlmapper-js). However, we do want to allow users to add custom functions for which we will need to save their code.